src/core/model_config.proto

```diff
@@ -713,6 +713,85 @@ message ModelQueuePolicy
 //@@
 message ModelDynamicBatching
 {
+  //@@  .. cpp:var:: message BatchInput
+  //@@
+  //@@     A batch input is an additional input for the requests in the batch
+  //@@     which describes the property of the request in the batch. The backend
+  //@@     should recognize batch input as ragged input.
+  //@@
+  message BatchInput
+  {
+    //@@
+    //@@    .. cpp:enum:: Kind
+    //@@
+    //@@       The kind of the batch input.
+    //@@
+    enum Kind {
+      //@@      .. cpp:enumerator:: Kind::BATCH_ELEMENT_COUNT = 0
+      //@@
+      //@@         The element count of the 'target_input' will be added as
+      //@@         input with shape [1].
+      //@@
+      BATCH_ELEMENT_COUNT = 0;
+
+      //@@      .. cpp:enumerator:: Kind::BATCH_ACCUMULATED_ELEMENT_COUNT = 1
+      //@@
+      //@@         The accumulated element count of the 'target_input' will be
+      //@@         added as input with shape [1]. For example, if there is a
+      //@@         batch of two request, each with 2 elements, an input of value
+      //@@         2 will be added to the first request, and an input of value
+      //@@         4 will be added to the second request.
+      //@@
+      BATCH_ACCUMULATED_ELEMENT_COUNT = 1;
+
+      //@@      .. cpp:enumerator::
+      //@@         Kind::BATCH_ACCUMULATED_ELEMENT_COUNT_WITH_ZERO = 2
+      //@@
+      //@@         The accumulated element count of the 'target_input' will be
+      //@@         added as input with shape [1], except for the first request
+      //@@         in the batch. For the first request in the batch, the input
+      //@@         will have shape [2] where the first element is value 0.
+      //@@
+      BATCH_ACCUMULATED_ELEMENT_COUNT_WITH_ZERO = 2;
+
+      //@@      .. cpp:enumerator:: Kind::BATCH_MAX_ELEMENT_COUNT_AS_SHAPE = 3
+      //@@
+      //@@         Among the requests in the batch, the max element count of the
+      //@@         'target_input' will be added as input with shape
+      //@@         [max_element_count] for the first request in the batch.
+      //@@         For other requests, such input will be with shape [0].
+      //@@         The data of the tensor will be dummy data.
```

Comment:
When we create this tensor we must provide some data buffer. Is is zero-filled or uninitialized?

Reply:
It is uninitialized currently. Plan backend just allocated a large enough buffer for this input on model creation.

Reply:
Can you change the comment to say "The data of the tensor will be uninitialized".